client/src/main/scala/ch.epfl.scala.index.client/CopyToClipboard.scala

```diff
@@ -0,0 +1,84 @@
+package ch.epfl.scala.index.client
+
+import scala.scalajs.js.annotation.JSExportTopLevel
+
+import org.scalajs.dom
+import org.scalajs.dom.document
+import org.scalajs.dom.raw.{Event, HTMLButtonElement, NodeList}
+
+object CopyToClipboard {
+  private val errorPrefix = "[CopyToClipboard] -"
+
+  /**
+   * Add listeners on the copy buttons
+   * @param triggerClass the HTML tag class identifying the buttons to listen
+   */
+  @JSExportTopLevel("addCopyListenersOnClass")
+  def addCopyListenersOnClass(triggerClass: String): Unit = {
+    val triggers: NodeList = document.getElementsByClassName(triggerClass)
+
+    for (i <- 0 until triggers.length) {
+      val trigger = triggers.item(i)
+      trigger.addEventListener(
+        `type` = "click",
+        listener = listener(targetId = "data-clipboard-target",
+                            displayMsg = Some("Copied!")),
+        useCapture = true
+      )
+    }
+  }
+
+  /**
+   * Listen a copy button
+   * @param targetId identifies the tag to be copied. Prefixed '#' is removed
+   * @param displayMsg Change the button text for 4s by the provided text
+   * @return
+   */
+  private def listener(targetId: String,
+                       displayMsg: Option[String]): Event => Unit = { event =>
+    event.target match {
+      case button: HTMLButtonElement =>
+        val targetAttribute = button.getAttribute(targetId).stripPrefix("#")
+        val target = document.getElementById(targetAttribute)
+
+        if (target != null) {
+          copyToClipboard(target.textContent)
+
+          displayMsg match {
+            case Some(tempMsg) =>
+              val previousContent = button.innerHTML
+              button.innerHTML = tempMsg
```

Comment:
What do you think of using `textContent` instead of `innerHTML`? `textContent` is a safer default, IMHO, since it prevents DOM injection.